modules/fedimint-dummy-common/src/db.rs

```diff
@@ -1,64 +1,107 @@
-/// ExampleKeyV0, then inserts the new ExampleKeys.
-pub async fn migrate_dummy_db_version_0<'a, 'b>(
-    dbtx: &'b mut DatabaseTransaction<'a>,
-) -> Result<(), anyhow::Error> {
-    let example_keys_v0 = dbtx
-        .find_by_prefix(&ExampleKeyPrefixV0)
-        .await
-        .collect::<Vec<_>>()
-        .await;
-    dbtx.remove_by_prefix(&ExampleKeyPrefixV0).await;
-    for (key, val) in example_keys_v0 {
-        let key_v2 = ExampleKey(key.0, "Example String".to_string());
-        dbtx.insert_new_entry(&key_v2, &val).await;
-    }
-    Ok(())
-}
+use crate::{DummyOutputOutcome, DummyPrintMoneyRequest};
 
+/// Namespaces DB keys for this module
 #[repr(u8)]
 #[derive(Clone, EnumIter, Debug)]
 pub enum DbKeyPrefix {
-    Example = 0x80,
+    DummyFunds = 0x01,
+    DummyOutcome = 0x02,
+    DummyPrint = 0x03,
 }
 
+// TODO: Boilerplate-code
 impl std::fmt::Display for DbKeyPrefix {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         write!(f, "{self:?}")
     }
 }
 
+/// Example old version 0 of DB entries
+// TODO: can we simplify this by just using macros?
 #[derive(Debug, Clone, Encodable, Decodable, Eq, PartialEq, Hash, Serialize)]
-pub struct ExampleKeyV0(pub u64);
+pub struct DummyFundsKeyV0(pub XOnlyPublicKey);
 
 #[derive(Debug, Encodable, Decodable)]
-pub struct ExampleKeyPrefixV0;
+pub struct DummyFundsKeyPrefixV0;
 
 impl_db_record!(
-    key = ExampleKeyV0,
-    value = (),
-    db_prefix = DbKeyPrefix::Example,
+    key = DummyFundsKeyV0,
+    value = Amount,
+    db_prefix = DbKeyPrefix::DummyFunds,
+);
+impl_db_lookup!(key = DummyFundsKeyV0, query_prefix = DummyFundsKeyPrefixV0);
+
+/// Example DB migration from version 0 to version 1
+pub async fn migrate_to_v1(dbtx: &mut DatabaseTransaction<'_>) -> Result<(), anyhow::Error> {
+    // Select old entries
+    let v0_entries = dbtx
+        .find_by_prefix(&DummyFundsKeyPrefixV0)
+        .await
+        .collect::<Vec<(DummyFundsKeyV0, Amount)>>()
+        .await;
+
+    // Remove old entries
+    dbtx.remove_by_prefix(&DummyFundsKeyPrefixV0).await;
+
+    // Migrate to new entries
+    for (v0_key, v0_val) in v0_entries {
+        let v1_key = DummyFundsKeyV1(v0_key.0);
+        dbtx.insert_new_entry(&v1_key, &v0_val).await;
+    }
+    Ok(())
+}
+
+/// Lookup tx outputs by key or prefix
+#[derive(Debug, Clone, Encodable, Decodable, Eq, PartialEq, Hash, Serialize)]
+pub struct DummyOutcomeKeyV1(pub OutPoint);
+
+#[derive(Debug, Encodable, Decodable)]
+pub struct DummyOutcomeKeyV1Prefix;
+
+impl_db_record!(
+    key = DummyOutcomeKeyV1,
+    value = DummyOutputOutcome,
+    db_prefix = DbKeyPrefix::DummyOutcome,
+);
+impl_db_lookup!(
+    key = DummyOutcomeKeyV1,
+    query_prefix = DummyOutcomeKeyV1Prefix
+);
+
+/// Lookup funds for a user by key or prefix
+#[derive(Debug, Clone, Encodable, Decodable, Eq, PartialEq, Hash, Serialize)]
+pub struct DummyFundsKeyV1(pub XOnlyPublicKey);
```

Comment:
Does `DummyFundsKeyV1` have the same data as `DummyFundsKeyV0`? Can we make them different so that the migration actually changes the data structure?

Reply:
Sure, it's mostly for an example anyway.